src/Compilers/VisualBasic/Portable/Binding/Binder_Lookup.vb

```diff
@@ -430,6 +450,29 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Next
             End Sub
 
+            Shared Function GetTypeForIntrinsicAlias(possibleAlias As String) As SpecialType
+                Dim aliasAsKeyword As SyntaxKind = SyntaxFacts.GetKeywordKind(possibleAlias)
+
+                Select Case aliasAsKeyword
+                    Case SyntaxKind.DateKeyword
+                        Return SpecialType.System_DateTime
+                    Case SyntaxKind.UShortKeyword
+                        Return SpecialType.System_UInt16
+                    Case SyntaxKind.ShortKeyword
+                        Return SpecialType.System_Int16
+                    Case SyntaxKind.UIntegerKeyword
+                        Return SpecialType.System_UInt32
+                    Case SyntaxKind.IntegerKeyword
+                        Return SpecialType.System_Int32
+                    Case SyntaxKind.ULongKeyword
+                        Return SpecialType.System_UInt64
+                    Case SyntaxKind.LongKeyword
+                        Return SpecialType.System_Int64
```

Comment:
Because their intrinsic name and actual type name are the same, therefore, no special handling needed.